web-app/HeimScripts/heatmap/run.R

```diff
@@ -54,31 +59,85 @@ main <- function(max_rows = 100, sorting = "nodes") {
     df <-
-      applySorting(df,sorting)  # no need for sorting in one sample case
-    variances <-
-      apply(df[,3:ncol(df)],1,var, na.rm = T)  # Calculating
-    # variance per probe (per row)
-    means <- rowMeans(df[,3:ncol(df)], na.rm = T)  # this is just an
-                                                   # auxiliary column - it will not be
-                                                   # used for JSON.
-    sdses <- apply(df[,3:ncol(df)],1,sd, na.rm = T)  # this is just
-                                                     # an auxiliary column -
-                                                     # it will not be used for JSON.
-    df["MEAN"] <- means
-    df["SD"] <- sdses
-    df["SIGNIFICANCE"] <- variances
-    df <- df[with(df, order(-SIGNIFICANCE)),]
+      applySorting(df,sorting)  # no need for sorting in one sample case, we do it here only
+    if ( twoSubsets ){
+      markerTable  <- getDEgenes(df)  # relies on columns being sorted,
+                                      # with subset1 first, this is because of the way
+    } else {                          # design matrix is being constructed
+      cleanUpLimmaOutput()  # In order to prevent displaying the table from previous run.
+    }
+    useLimma <- !is.function(rankingMethod)  # rankingMethod is either a function or a character "limma"
+    if (useLimma  && !twoSubsets) {  # cannot use any of the limma methods for single subset.
+      stop( paste("Illegal ranking method: ", ranking, " two subsets needed.") )
+    }
+    if (useLimma ) {
+      rankingScore <- markerTable[ranking]
+      if (length(rankingScore) == 1 && is.na(rankingScore[1])) {
+           stop(paste("Illegal ranking method selected: ",ranking))
+        }
+    } else {
+      cleanUpLimmaOutput()  # Delete json file so that it is not displayed
+      rankingScore <-
+        apply(measurements, 1, rankingMethod)  # Calculating
+    }                                          # ranking per probe (per row)
+    means <- rowMeans(measurements, na.rm = T)  # this is just an
+                                                # auxiliary column - it will not be
+                                                # used for JSON.
+    sdses <- apply(measurements,1 ,sd , na.rm = T)  # this is just
+                                                    # an auxiliary column -
+                                                    # it will not be used for JSON.
+    df["MEAN"]         <- means
+    df["SD"]           <- sdses
+    df["SIGNIFICANCE"] <- rankingScore
+    df                 <- df[with(df, order(-SIGNIFICANCE)), ]
+    if (twoSubsets) {
+      markerTable["SIGNIFICANCE"] <- rankingScore
+      markerTable                 <- markerTable[with(markerTable, order(-SIGNIFICANCE)), ]
+      markerTable["SIGNIFICANCE"] <- NULL
+      markerTable                 <- markerTable[1:min(max_rows, nrow(markerTable)), ]
+      writeMarkerTable(markerTable)
+    }
   }
-  else{
+  else {
```

Comment:
no row selection  if there's only one sample? In that case all rows are displayed? Or will a heatmap not be made anyway?